docs/manual/src/internals/rendering_foreign_bindings.md

```diff
@@ -0,0 +1,82 @@
+# Rendering Foreign Bindings
+
+This document details the general system that UniFFI uses to render the foreign bindings code.
+
+## The Askama template engine
+
+Our foreign bindings generation is based off the [Askama](https://djc.github.io/askama/) template rendering engine.
+Askama uses a compile-time macro system that allows the template code to use Rust types directly, calling their methods
+passing them to normal Rust functions.
+
```

Comment:
It would be nice to provide some transitionary overview. Something between "We use Askama" and "Talk about mega-matches".

e.g. taking a `ComponentInterface`, and rendering the types into strings, and then assembling those strings into a single file.

and: we do this like this because…